src/testing/flow_solver.cpp

```diff
@@ -1,35 +1,324 @@
 #include "flow_solver.h"
 
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <stdlib.h>
+#include <vector>
+#include <sstream>
+
 namespace PHiLiP {
 
 namespace Tests {
 //=========================================================
 // FLOW SOLVER TEST CASE -- What runs the test
 //=========================================================
 template <int dim, int nstate>
-FlowSolver<dim, nstate>::FlowSolver(const PHiLiP::Parameters::AllParameters *const parameters_input, std::shared_ptr<FlowSolverCaseBase<dim, nstate>> flow_solver_case)
+FlowSolver<dim, nstate>::FlowSolver(
+    const PHiLiP::Parameters::AllParameters *const parameters_input, 
+    std::shared_ptr<FlowSolverCaseBase<dim, nstate>> flow_solver_case,
+    const dealii::ParameterHandler &parameter_handler_input)
 : TestsBase::TestsBase(parameters_input)
 , flow_solver_case(flow_solver_case)
+, parameter_handler(parameter_handler_input)
 , initial_condition_function(InitialConditionFactory<dim,double>::create_InitialConditionFunction(parameters_input, nstate))
 , all_param(*parameters_input)
 , flow_solver_param(all_param.flow_solver_param)
 , ode_param(all_param.ode_solver_param)
 , poly_degree(all_param.grid_refinement_study_param.poly_degree)
 , final_time(flow_solver_param.final_time)
+, input_parameters_file_reference_copy_filename(flow_solver_param.restart_files_directory_name + std::string("/") + std::string("input_copy.prm"))
 , dg(DGFactory<dim,double>::create_discontinuous_galerkin(&all_param, poly_degree, flow_solver_case->generate_grid()))
 , ode_solver(ODE::ODESolverFactory<dim, double>::create_ODESolver(dg))
 {
     dg->allocate_system();
     flow_solver_case->display_flow_solver_setup();
-    pcout << "Initializing solution with initial condition function..." << std::flush;
+    
     dealii::LinearAlgebra::distributed::Vector<double> solution_no_ghost;
     solution_no_ghost.reinit(dg->locally_owned_dofs, MPI_COMM_WORLD);
-    dealii::VectorTools::interpolate(dg->dof_handler, *initial_condition_function, solution_no_ghost);
-    dg->solution = solution_no_ghost;
+
+    if(flow_solver_param.restart_computation_from_file == true) {
+        if(dim == 1) {
+            pcout << "Error: restart_computation_from_file is not possible for 1D. Set to false." << std::endl;
+            std::abort();
+        }
+#if PHILIP_DIM>1
```

Comment:
I'd guess this has to do with SolutionTransfer not existing for 1D?

Move it such that only the part that is dim-dependent is in your compiler #if .

Reply:
addressed in most recent commit